apiserver/facades/agent/uniter/register.go

```diff
@@ -20,12 +20,12 @@ import (
 // Register is called to expose a package of facades onto a given registry.
 func Register(registry facade.FacadeRegistry) {
 	registry.MustRegister("Uniter", 18, func(ctx facade.Context) (facade.Facade, error) {
-		return newUniterAPI(ctx)
+		return newUniterAPI(ctx, ctx.ServiceFactory().ControllerConfig())
 	}, reflect.TypeOf((*UniterAPI)(nil)))
 }
 
 // newUniterAPI creates a new instance of the core Uniter API.
-func newUniterAPI(context facade.Context) (*UniterAPI, error) {
+func newUniterAPI(context facade.Context, controllerConfigGetter ControllerConfigGetter) (*UniterAPI, error) {
```

Comment:
Previously, we did it in a bit different way:

- we define `New<facadeName>Facade` func that takes facade.Context
- than we define `New<facadeName>API` that already takes all needed info as a parameters

The current approach looks like something hybrid, and I don't want to use different patterns for different facades.

Reply:
For now, I'm starting to boil the ocean here with all the changes. And I'm starting to worry about all the changes actually breaking something. So I'd rather just do things incrementally, so I can actually pinpoint what changed and how. Doing too much at once removes that chance to catch it.